### ISSUE TITLE
Using Datagrid, when there's many headers, footer / pagination gets pushed off page

### ISSUE THREAD

**Issue by spetri** [vmware/clarity#1677]
**Select one ...**  (check one with "x")
```
[x ] bug
[ ] feature request
[ ] enhancement
```

### Expected behavior
When I see a data-grid on the page with many headers (i.e. more than 10-15 so that the table goes off screen horizontally), I should still be able to paginate between pages without scrolling.

### Actual behavior
When I see a data-grid on the page with many headers (i.e. more than 10-15 so that the table goes off screen horizontally), the page footer also gets stretched, so the user doesn't see the pagination information unless they scroll.

### Reproduction of behavior
https://plnkr.co/edit/8FuGECUhfHX83iXe2NIn?p=preview

### Workaround
In the `data-grid.component.css` file:
```
/deep/ .datagrid {
  max-width: 100%;
  .datagrid-table-wrapper {
    overflow: auto;
  }
  .datagrid-head, .datagrid-body {
    display: table;
    width: 100%;
  }
}
```
The solution becomes: 
![screen shot 2017-11-07 at 1 44 49 pm](https://user-images.githubusercontent.com/1146420/32519514-e10a9920-c3c1-11e7-9771-5076be5bf006.png)

And seems to work for us!

* **Angular version:** 4.4.5

* **Clarity version:**
Latest version

* **OS and version:** 
MacOSX latest

* **Browser:** [all]

**Comment by rbirkgit** [comment]
The workaround above causes new issues (at least with 0.10.14). We get unnecessary horizontal scrollbar as grid gets too wide many times. e.g. if you have an 'expand' column, then grid gets that columns width extra wide. There are other cases too, but this one most easy to reproduce as it always happen.

The empty grid text is also no longer vertically centered. Small bug, but it's another side effect.

We got these two issues reported as bugs and with that can no longer use the workaround. I really hope there is another solution as most of our grids gets their pager/item number and vertical scrollbar hidden with the current Clarity.

**Comment by ph55** [comment]
Another workaround that did the trick for me.

```scss
  clr-datagrid {

    height: auto !important;
    margin-bottom: 1.5rem;
    position: relative;

    clr-dg-footer {
      position: absolute;
      bottom: -1.5rem;
      left: 0;
      right: 0;
    }
}
```

**Comment by youdz** [comment]
Fixed by #2670.
